### PR TITLE
Allow query of sample hierarchy

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

File: .github/workflows/nexus-publish-snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+          settings-path: ${{ github.workspace }}
```
Comment (Steffengreiner) [inline .github/workflows/nexus-publish-snapshots.yml:23]: Is this parser written in java 11?

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```
Comment (Steffengreiner) [inline .github/workflows/run-tests.yml:20]: Is this parser written in java 11?

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+          settings-path: ${{ github.workspace }}
```
Comment (Steffengreiner) [inline .github/workflows/create-release.yml:19]: Is this parser written in java 11?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+        xmlns="http://maven.apache.org/POM/4.0.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>life.qbic</groupId>
+    <artifactId>openbis-scripts</artifactId>
+    <version>1.0.0</version>
+    <name>Postman cli</name>
```
Comment (Steffengreiner) [inline pom.xml]: Why is this postman-cli?

File: src/main/java/life/qbic/App.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.logging.log4j.Logger;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * postman for staging data from openBIS
+ */
```
Comment (Steffengreiner) [inline src/main/java/life/qbic/App.java]: Why is this called postman? 🤔

File: src/main/java/life/qbic/io/commandline/SampleHierarchyCommand.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  AuthenticationOptions auth = new AuthenticationOptions();
+
+    @Override
+    public void run() {
+        List<String> spaces = new ArrayList<>();
+        if(space!=null) {
+          spaces.add(space);
+        }
+        Authentication authentication = App.loginToOpenBIS(auth.getPassword(), auth.user, auth.as_url);
+        OpenbisConnector openbis = new OpenbisConnector(auth.as_url, authentication.getSessionToken(), "/Users/afriedrich/Downloads/");
```
Comment (Steffengreiner) [inline src/main/java/life/qbic/io/commandline/SampleHierarchyCommand.java]: I think this should not be hardcoded 🤔

File: src/main/java/life/qbic/model/Configuration.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import java.nio.file.Paths;
+
+/**
+ * Global configuration container
+ */
+public class Configuration {
+
+    public static final long MAX_DOWNLOAD_ATTEMPTS = 3;
+    public static final Path LOG_PATH = Paths.get(System.getProperty("user.dir"),"logs");
+}
```
Comment (Steffengreiner) [inline src/main/java/life/qbic/model/Configuration.java:13]: Does this save the logs into the applications directory or the user directory (e.g. afriedrich)?

File: src/main/java/life/qbic/model/download/FileSystemWriter.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    this.matchingSummaryFile = new File(matchingSummaryFile.toString());
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void reportSummary(List<String> summary) {
+    //TODO
+  }
```
Comment (Steffengreiner) [inline src/main/java/life/qbic/model/download/FileSystemWriter.java]: Will this be offered at some point? I'd argue to remove this method if it's not actively used 🤔

File: src/main/java/life/qbic/model/download/OutputPathFinder.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        Path finalPath = Paths.get("");
+        if (outputPath != null && !outputPath.isEmpty()) {
+            if(isPathValid(outputPath)) {
+                finalPath = Paths.get(outputPath + path);
+            } else{
+                LOG.error("The path you provided does not exist.");
+                System.exit(1);
+            }
+        } else {
+            finalPath = Paths.get(System.getProperty("user.dir") + path);
```
Comment (Steffengreiner) [inline src/main/java/life/qbic/model/download/OutputPathFinder.java:79]: Does this save the logs into the applications directory or the user directory (e.g. afriedrich)?

File: src/main/resources/log4j2.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="INFO">
+  <Appenders>
+    <Console name="Console" target="SYSTEM_OUT">
+      <PatternLayout pattern="[%d{HH:mm:ss.SSS}] %msg%n"/>
+    </Console>
+    <File append="true" fileName="logs/postman-all.log" immediateFlush="false" name="MyFile">
```
Comment (Steffengreiner) [inline src/main/resources/log4j2.xml]: This should not be the postman-all log i think

**Review by Steffengreiner — CHANGES_REQUESTED** [review]
Heya @wow-such-code Awesome work, however I have some questions concerning the naming schema and the java version used within the github actions

File: src/main/java/life/qbic/model/Configuration.java (hunk 7 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/model/Configuration.java:13]: user.dir is the current directory

File: src/main/java/life/qbic/model/download/OutputPathFinder.java (hunk 9 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/model/download/OutputPathFinder.java:79]: user.dir is the current directory

**Review by Steffengreiner — APPROVED** [review]
Thanks for addressing the changes as requested :)

**Pull request merged**